Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -713,6 +731,18 @@ int HGCalDDDConstants::wafers() const {
   return wafer;
 }
 
+int HGCalDDDConstants::wafers(int layer, int type) const {
+
+  int  wafer(0);
+  auto itr = waferLayer_.find(layer);
+  if (itr != waferLayer_.end()) {
```

Comment:
I'm not following - you are doing
if ( type==2 ) wafer=(itr->second)[2]
if ( type==1 ) wafer=(itr->second)[1]
etc..